### PR TITLE
Fix CR-1119807 - handle drm callbacks for non cma memory in edge platforms

### PR DESCRIPTION
**Pull request by rbramand-xilinx**
#### Problem solved by the commit
In edge platforms drm callbacks handle only cma memory, for non cma memory these callbacks are not working properly. This PR fixes some of those callbacks.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
While handling [CR-1119807](https://jira.xilinx.com/browse/CR-1119807) this bug was discovered.

#### How problem was solved, alternative solutions (if any) and why they were rejected
For CMA buffer we are appending extra flag to indicate memory is allocated in CMA region, checking that flag to determine between cma vs non-cma

#### Risks (if any) associated the changes in the commit
Minimal testing is done, at present we don't have testcases to verify all the possible combinations

#### What has been tested and how, request additional testing if necessary
Tested locally with dpu application that uses mmap on exportBO and the test is passing with both cma and non-cma memory. Also used XRT pipeline to verify.

#### Documentation impact (if any)
None

### CODE REVIEW

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 1]
```diff
@@ -814,6 +814,56 @@ static int zocl_iommu_init(struct drm_zocl_dev *zdev,
 	return 0;
 }
 
+static int zocl_gem_prime_mmap(struct drm_gem_object *obj,
+		struct vm_area_struct *vma)
+{
+	struct drm_zocl_bo *zocl_obj;
```
Comment (saifuddin-xilinx) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c:820]: Make it oneline like other place.
i.e. struct drm_zocl_bo *zocl_obj = to_zocl_bo(obj);

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 2]
```diff
@@ -814,6 +814,56 @@ static int zocl_iommu_init(struct drm_zocl_dev *zdev,
...
+
+	if (zocl_obj->flags & ZOCL_BO_FLAGS_CMA)
+		return drm_gem_cma_prime_vmap(obj);
+	else
+		return zocl_obj->vmapping;
+}
+
+void zocl_gem_prime_vunmap(struct drm_gem_object *obj, void *vaddr)
+{
+
```
Comment (saifuddin-xilinx) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c:846]: Add comments and add a return statement.

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 3]
```diff
@@ -814,6 +814,56 @@ static int zocl_iommu_init(struct drm_zocl_dev *zdev,
...
+		return drm_gem_cma_vmap(obj, map);
+	else {
+		dma_buf_map_set_vaddr(map, zocl_obj->vmapping);
+		return 0;
+	}
+}
+
+void zocl_gem_prime_vunmap(struct drm_gem_object *obj, struct dma_buf_map *map)
+{
+
```
Comment (saifuddin-xilinx) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c:863]: Add comments and add a return statement.

**Comment by rbramand-xilinx** [comment]
closing this PR as we need to discuss more on how to handle non cma memory call backs

**Pull request closed without merging**